src/modelinstance.cpp

```diff
@@ -751,36 +766,12 @@ Status ModelInstance::fetchModelFilepaths() {
         return StatusCode::PATH_INVALID;
     }
 
-    bool found = true;
-    for (auto extension : OV_MODEL_FILES_EXTENSIONS) {
-        auto file = findModelFilePathWithExtension(extension);
-        if (file.empty()) {
-            found = false;
-        }
-        modelFiles.push_back(file);
-    }
-    if (!found) {
-        found = true;
-        modelFiles.clear();
-        for (auto extension : ONNX_MODEL_FILES_EXTENSIONS) {
-            auto file = findModelFilePathWithExtension(extension);
-            if (file.empty()) {
-                found = false;
-            }
-            modelFiles.push_back(file);
-        }
-    }
-    if (!found) {
-        found = true;
-        modelFiles.clear();
-        for (auto extension : PADDLE_MODEL_FILES_EXTENSIONS) {
-            auto file = findModelFilePathWithExtension(extension);
-            if (file.empty()) {
-                found = false;
-            }
-            modelFiles.push_back(file);
-        }
-    }
+    bool found = false;
+    fetchModelFiles(&found, OV_MODEL_FILES_EXTENSIONS);
+    fetchModelFiles(&found, ONNX_MODEL_FILES_EXTENSIONS);
+    fetchModelFiles(&found, PADDLE_MODEL_FILES_EXTENSIONS);
+    fetchModelFiles(&found, TF_MODEL_FILES_EXTENSIONS);
```

Comment:
Maybe we could loop through accepted formats? Something like:
```
for (i = 0; i < sizeof(ACCEPTED_FORMATS) && !found; i++) {
    fetchModelFiles(found, ACCEPTED_FORMATS[i]);
}
```
What do you think?